src/Cryptography/Sodium/Bindings/Scrypt.hs

```diff
@@ -0,0 +1,253 @@
+-- logging in on a website), a 1 second computation is likely to be the acceptable maximum. For interactive use (e.g. a desktop application), a 5 second
+-- pause after having entered a password is acceptable if the password doesn't need to be entered more than once per session. For non-interactive and
+-- infrequent use (e.g. restoring an encrypted backup), an even slower computation can be an option. However, the best defense against brute-force password
+-- cracking is to use strong passwords. Libraries such as [passwdqc](https://www.openwall.com/passwdqc/) can help enforce this.
+--
+-- == Notes
+--
+-- Do not use constants to verify a password or produce a deterministic output.
+-- Save the parameters alongside the hash instead.
+-- By doing so, passwords can be rehashed using different parameters if required later on.
+--
+-- By design, a password whose length is 65 bytes or more is reduced to SHA-256(password).
+-- This can have security implications if the password is present in another password database
+-- using raw, unsalted SHA-256 or when upgrading passwords previously hashed with unsalted SHA-256 to scrypt.
+--
+-- It is highly recommended to use 'Cryptography.Sodium.Bindings.SecureMemory.lock'
+-- to lock memory regions storing plaintext passwords and to call
+-- 'Cryptography.Sodium.Bindings.SecureMemory.unlock' right after
+-- 'cryptoPWHashScryptSalsa2018SHA256Str' and 'cryptoPWHashScryptSalsa2018SHA256StrVerify'
+-- return.
+
+-- | Derive a key from a password.
+--
+-- For interactive, online operations, 'cryptoPWHashScryptSalsa2018SHA256OpsLimitInteractive'
+-- and 'cryptoPWHashScryptSalsa2018SHA256MemLimitInteractive' provide a safe baseline to be used.
+-- However, using higher values may improve security.
+--
+-- For highly sensitive data, 'cryptoPWHashScryptSalsa2018SHA256OpsLimitSensitive' and
+-- 'cryptoPWHashScryptSalsa2018SHA256MemLimitSensitive' can be used as an alternative.
+-- However, with these parameters, deriving a key takes about
+-- 2 seconds on a 2.8 GHz Core i7 CPU and requires up to 1 GiB of dedicated RAM.
+--
+-- The salt should be unpredictable. 'Cryptography.Sodium.Bindings.Random.randombytesBuf'
+-- is the easiest way to fill the 'cryptoPWHashScryptSalsa2018SHA256SaltBytes' bytes
+-- of the salt.
+--
+-- Keep in mind that to produce the same key from the same password, the same salt,
+-- opslimit, and memlimit values must be used. Therefore, these parameters must be stored for each user.
+--
+-- /See also:/ [crypto_pwhash_scryptsalsa208sha256()](https://doc.libsodium.org/advanced/scrypt#key-derivation)
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_pwhash_scryptsalsa208sha256"
+  cryptoPWHashScryptSalsa2018SHA256 ::
+    -- | A pointer to the computed key.
+    Ptr CUChar ->
+    -- | The length of the computer key.
+    -- Should be comprised between 'cryptoPWHashScryptSalsa2018SHA256BytesMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256BytesMax' (~127 GB).
```

Comment:
'Comprised' is unnecessary here. "Should be between [something] and [something else]" is fine.